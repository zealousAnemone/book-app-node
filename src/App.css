.centered {
  text-align: center;
}

.header {
  font-weight: 700;
}
.author-name {
  display: inline-block;
  width: 40%;
}

.mark-dupe {
  display: inline-block;
  width: 5%;
}
#dupe-popup {
  border: 2px solid #444;
  border-radius: 10px;
  box-shadow: 3px 3px 3px #444;
  padding: 30px;
  text-align: center;
  width: 40%;
  margin: 200px auto;
}

#dupe-popup button {
  margin-right: 5px;
}

#dupe-popup input {
  margin-right: 2em;
}

.blurred {
  filter: blur(2px);
  -webkit-filter: blur(2px);
}
.book-name {
  display: inline-block;
  width: 40%;
  margin-right: 5%;
}

.author-name:hover,
.book-name:hover {
  font-weight: 500;
}
ul {
  border: 2px solid #444;
  border-radius: 10px;
  margin: 40px;
  padding: 30px;
}

li {
  list-style: none;
  padding: 14px;
}

li:hover {
  cursor: pointer;
}

li:nth-child(odd) {
  background: #ffe4e1;
}
