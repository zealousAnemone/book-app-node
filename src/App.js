import './App.css';
import { useState, useEffect } from 'react';
import DupePopup from './components/DupePopup';

function App() {
  const [books, setBooks] = useState([]);
  const [singleBook, setSingleBook] = useState([]);
  const [bookView, toggleBookView] = useState(false);
  const [authorBooks, setAuthorBooks] = useState([]);
  const [authorView, toggleAuthorView] = useState(false);
  const [dupeView, toggleDupeView] = useState(false);

  let bookList = [];
  let authorList = [];

  const displayAuthor = (name) => {
    let author = { name };
    fetch('/author', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(author),
    })
      .then((res) => res.json())
      .then((authBooks) => {
        setAuthorBooks(authBooks);
        toggleAuthor();
      });
  };

  const markDupe = (dupe, original) => {
    console.log('duplicate!');
  };

  const displayBook = (title) => {
    let bookTitle = { title };
    console.log('displayBook: ', bookTitle);
    fetch('/singlebook', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(bookTitle),
    })
      .then((res) => res.json())
      .then((singleBook) => {
        setSingleBook(singleBook);
        toggleBook();
        console.log(singleBook);
      });
  };

  const toggleAuthor = () => {
    toggleAuthorView(!authorView);
  };

  const toggleBook = () => {
    toggleBookView(!bookView);
  };

  const toggleDupe = () => {
    toggleDupeView(!dupeView);
    console.log(dupeView);
  };

  useEffect(() => {
    fetch('/books')
      .then((res) => res.json())
      .then((books) => {
        setBooks(books);
      });
  }, []);

  for (let i = 0; i < authorBooks.length; i++) {
    authorList.push(<li key={i}>{authorBooks[i].title}</li>);
  }
  for (let i = 0; i < books.length; i++) {
    bookList.push(
      <li key={i}>
        <span
          className="book-name"
          id={books[i].title}
          onClick={(e) => {
            displayBook(e.target.id);
          }}
        >
          {books[i].title}
        </span>
        <span
          className="author-name"
          id={books[i].name}
          onClick={(e) => {
            displayAuthor(e.target.id);
          }}
        >
          {books[i].name}
        </span>
        <input
          type="checkbox"
          id={books[i].title}
          onClick={() => toggleDupe()}
        ></input>
      </li>
    );
  }
  return (
    <div className="App">
      {!authorView && !bookView && !dupeView && (
        <ul>
          <li>
            <span className="header">
              <span className="book-name">Title</span>
              <span className="author-name">Author</span> Duplicate?
            </span>
          </li>
          {bookList}
        </ul>
      )}
      {authorView && (
        <div className="centered">
          <h1>Books by {authorBooks[0].name}</h1>
          <ul>{authorList}</ul>
          <button onClick={toggleAuthor}>Back to List</button>
        </div>
      )}
      {bookView && (
        <div className="centered">
          <h1>{singleBook[0].title}</h1>
          <h2>{singleBook[0].name}</h2>
          <button onClick={toggleBook}>Back to List</button>
        </div>
      )}
      {dupeView && <DupePopup />}
    </div>
  );
}

export default App;
