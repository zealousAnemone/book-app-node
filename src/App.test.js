// import { render, screen } from '@testing-library/react';
// import App from './App';

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  it('renders on page load', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });
});
