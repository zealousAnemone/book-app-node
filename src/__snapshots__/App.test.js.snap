// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders on page load 1`] = `
<div
  className="App"
>
  <div>
    <ul>
      <li>
        <span
          className="header"
        >
          <span
            className="book-name"
          >
            Title
          </span>
          <span
            className="author-name"
          >
            Author
          </span>
           
          <span
            className="mark-dupe"
          >
            Duplicate?
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`div with className App is rendered on page load 1`] = `
<div
  className="App"
>
  <div>
    <ul>
      <li>
        <span
          className="header"
        >
          <span
            className="book-name"
          >
            Title
          </span>
          <span
            className="author-name"
          >
            Author
          </span>
           
          <span
            className="mark-dupe"
          >
            Duplicate?
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
